### PR TITLE
Add missing monitor options

### PR DESCRIPTION
**Pull request by HantingZhang2**
### What does this PR do?
This PR adds the following missing monitor options: 

GroupRetentionDuration
NotificationPresetName
NotifyBy
OnMissingData
SchedulingOptions

### Review checklist (to be filled by reviewers)

- [ ] Feature or bug fix MUST have appropriate tests (unit, integration, etc...)
- [ ] PR title must be written as a CHANGELOG entry [(see why)](/CONTRIBUTING.md#pull-request-title)
- [ ] Files changes must correspond to the primary purpose of the PR as described in the title (small unrelated changes should have their own PR)
- [ ] PR must have one `changelog/` label attached. If applicable it should have the `backward-incompatible` label attached.
- [ ] PR should not have `do-not-merge/` label attached.
- [ ] If Applicable, issue must have `kind/` and `severity/` labels attached at least.

### CODE REVIEW

File: datadog-monitors-monitor-handler/inputs/inputs_1_create.json [hunk 1]
```diff
@@ -9,9 +9,17 @@
...
+    "NotificationPresetName":"show_all",
     "NotifyNoData": false,
     "RenotifyInterval": 0,
     "EnableLogsSample": true,
+    "SchedulingOptions": {
+			"EvaluationWindow": {
+				"DayStarts": "22:00",
+        "MonthStarts":1
+			}
+    },
```
Comment (skarimo) [inline datadog-monitors-monitor-handler/inputs/inputs_1_create.json:22]: Please format these

File: datadog-monitors-monitor-handler/inputs/inputs_1_update.json [hunk 2]
```diff
@@ -9,9 +9,16 @@
...
+    "NotifyBy": ["host"],
+    "NotificationPresetName":"show_all",
     "NotifyNoData": true,
     "RenotifyInterval": 5,
     "EnableLogsSample": true,
+    "SchedulingOptions": {
+			"EvaluationWindow": {
+				"HourStarts":1
+			}
+    },
```
Comment (skarimo) [inline datadog-monitors-monitor-handler/inputs/inputs_1_update.json:21]: Same here

File: datadog-monitors-monitor-handler/src/datadog_monitors_monitor/handlers.py [hunk 3]
```diff
@@ -27,6 +29,8 @@
 from datadog_api_client.v1.model.monitor_formula_and_function_events_data_source import (
     MonitorFormulaAndFunctionEventsDataSource as ApiMonitorMonitorFormulaAndFunctionEventsDataSource,
 )
+from datadog_api_client.v1.model.on_missing_data_option import OnMissingDataOption
+from datadog_api_client.v1.model.monitor_options_notification_presets import MonitorOptionsNotificationPresets
```
Comment (skarimo) [inline datadog-monitors-monitor-handler/src/datadog_monitors_monitor/handlers.py:33]: Lets import these as Api<ModelName> so we can more easily differentiate between client models vs cfn.

File: datadog-monitors-monitor-handler/src/datadog_monitors_monitor/handlers.py [hunk 4]
```diff
@@ -137,19 +143,26 @@ def read_handler(
     options = monitor.options if hasattr(monitor, "options") else None
     if options:
         model.Options = MonitorOptions(
-            EnableSamples=options.enable_samples if hasattr(options, "enable_samples") else None,
+            EnableSamples = options.enable_samples if hasattr(options, "enable_samples") else None,
```
Comment (skarimo) [inline datadog-monitors-monitor-handler/src/datadog_monitors_monitor/handlers.py:146]:
```suggestion
            EnableSamples=options.enable_samples if hasattr(options, "enable_samples") else None,
```

File: datadog-monitors-monitor-handler/src/datadog_monitors_monitor/handlers.py [hunk 5]
```diff
@@ -137,19 +143,26 @@ def read_handler(
...
+            GroupRetentionDuration=options.group_retention_duration if hasattr(options,"group_retention_duration") else None,
             IncludeTags=options.include_tags if hasattr(options, "include_tags") else None,
             Locked=options.locked if hasattr(options, "locked") else None,
             MinLocationFailed=options.min_location_failed if hasattr(options, "min_location_failed") else None,
             NewHostDelay=options.new_host_delay if hasattr(options, "new_host_delay") else None,
             NoDataTimeframe=options.no_data_timeframe if hasattr(options, "no_data_timeframe") else None,
             NotifyAudit=options.notify_audit if hasattr(options, "notify_audit") else None,
+            NotifyBy=[str(notify) for notify in options.notify_by]
+            if hasattr(options,"notify_by")
+            else None,
```
Comment (skarimo) [inline datadog-monitors-monitor-handler/src/datadog_monitors_monitor/handlers.py:159]:
```suggestion
            NotifyBy=[str(notify) for notify in options.notify_by] if hasattr(options, "notify_by") else None,
```

**Review by skarimo — COMMENTED** [review]
Looks great overall. Could you rebase on master and run pre-commit to format the code. Thanks!

File: datadog-monitors-monitor-handler/src/datadog_monitors_monitor/handlers.py (hunk 4 above)
Comment (HantingZhang2) [reply datadog-monitors-monitor-handler/src/datadog_monitors_monitor/handlers.py:146]: I think  `EnableSamples = options.enable_samples if hasattr(options, "enable_samples") else None` always had space for other options?

**Review by skarimo — APPROVED** [review]

**Pull request merged**